### DISCUSSION TITLE
`signInWIthOAuth` always redirects to `localhost:3000` in production

### DISCUSSION THREAD

**Discussion by hyochan** [supabase/supabase#12942] [Questions] [answered]
# Bug report

## Describe the bug

```
await supabase.auth.signInWithOAuth({
  provider: 'github',
  options: {
    redirectTo: 'https://google.com',
  },
});
```

I am using `nextjs13` and I am having trouble redirecting after signing from provider.
It always redirects to `localhost:3000` , so it works in developement but when I deployed to `vercel.com`, id doesn't work.

I tried `redirectTo` param in `options` but it doesn't work either.
I also set `NEXTAUTH_URL` after [reading stackoverflow](https://stackoverflow.com/questions/70717224/how-to-define-a-custom-base-path-when-nextauth-url-doesnt-work) but no luck.

## To Reproduce

You can try [sign in to github in dooboo.io](http://dooboo.io) which is my toy webapp.

## Expected behavior

Redirect to production url instead of `localhost:3000` after signing in to github.

## Screenshots

<img width="485" alt="Screenshot 2023-03-11 at 1 09 38 AM" src="https://user-images.githubusercontent.com/27461460/224366365-669f4fc3-b8c3-4101-b1e7-802ec616384f.png">
<img width="347" alt="Screenshot 2023-03-11 at 1 09 50 AM" src="https://user-images.githubusercontent.com/27461460/224366390-94ff5d1f-808a-4803-942a-06f0ea2a268c.png">

## System information

- OS: macOS
- Browser (if applies) chrome
- Version of supabase-js: [e.g. 6.0.2]
   ```
   "@supabase/auth-helpers-nextjs": "^0.5.6",
   "@supabase/auth-helpers-react": "^0.3.1",
   "@supabase/auth-ui-react": "^0.3.5",
   "@supabase/supabase-js": "^2.8.0",
   ```
- Version of Node.js: [e.g. 10.10.0]
   vercel default (nextjs13)

## Additional context

N/A

**Comment by GaryAustin1** [comment, marked as answer]
This seems like more a help request than a bug.  You would be better off seeking help in Supabase Discord or Discussions here first.

You likely did not set your exact URL's here:
![image](https://user-images.githubusercontent.com/54564956/224368331-1ffb974a-8e2f-47f5-be21-5975adb52a06.png)

So it will always to to the default siteURL setting.

**Reply by hyochan** [reply]
@GaryAustin1 Oh my god! This did the trick. I've been suffering for more than 2 hours with this. Should've asked earlier.

Thank you so much 👍

**Reply by anantdevcs** [reply]
+1 on the resolution, Thanks!

**Reply by Metehan-Altuntekin** [reply]
Thanks a lot, this saved me from looking for the fix in the wrong place

**Reply by kgrigor** [reply]
I am facing the same issue after adding the URLs.

**Reply by Metehan-Altuntekin** [reply]
@kgrigor Have you done it like `https://yourdomain.com/auth/callback`?

**Reply by alxvallejo** [reply]
@Metehan-Altuntekin What should be on the `auth/callback` route in your application if you're using an AuthProvider in React?

**Reply by eleijonmarck** [reply]
Configure it for your project - `https://supabase.com/dashboard/project/<project-id>/auth/url-configuration`

**Reply by alxvallejo** [reply]
No, that's the callback for the OAuth, not the redirect from a successful OAuth session creation. Previous comment shows "yourdomain.com", not supabase. So i'm wondering what they're doing to handle the successful session token and why that's not being handled by onAuthStateChange listener

**Reply by Metehan-Altuntekin** [reply]
@alxvallejo I think normal auth is handled by the Supabase package by setting token and `auth/callback` isn't needed for that. A redirect to home page would be good for UX though

**Reply by genthegreat** [reply]
Thanks for this. I spent 2 hours trying to figure out why my code wasn't working even after following the auth documentation.

**Comment by AndryRoby** [comment]
Matches what was already flagged in the answer: the `redirectTo` you pass in code is only ever a *request*. Supabase still checks it against the Redirect URLs allow-list in the dashboard, and if it's not there (or only the bare origin is listed, not the full path), it discards it and falls back to Site URL. That's why setting `redirectTo` correctly in `signInWithOAuth()` alone doesn't fix it, no matter how right the URL looks in your code. Worth adding for anyone finding this now: if you deploy preview branches too, those need their own wildcard allow-list entry (e.g. `https://*.vercel.app/**`) or they'll hit the exact same fallback.

I built a free client-side checker for exactly this failure mode. Paste your Site URL, allow-list and `redirectTo` and it flags the mismatch directly: https://arling.sk/supabase-redirect-doctor/
